torch/ao/sparsity/experimental/data_sparsifier/lightning/tests/test_callbacks.py

```diff
@@ -78,5 +105,121 @@ def test_post_training_callback(self):
         self.run_all_checks(pl_module, callback, sparsifier_args)
 
 
+class _TrainingAwareCallbackTestCase(TestCase):
+    """Class to test in-training version of lightning callback
+    Simulates model training and makes sure that each hook is doing what is expected
+    """
+    def _check_on_train_start(self, pl_module, callback, sparsifier_args, scheduler_args):
+
```

Comment:
would also be helpful if we can add some comments for these individual checks, describing in a high level what do we do here and what is the expected result